### PR TITLE
[REF-6034] MLOPs Metrics Beautification 1: Adding Classification/Regr…

### PR DESCRIPTION
**Pull request by harshilpm**
…ession StatObjectFactory

- Classes are responsible for creating stat object from given input.
- Removing big if elses and taking advantage of dicts.

Testing Done:
- Unit Tests

### CODE REVIEW

File: mlops/parallelm/mlops/metrics_constants.py [hunk 1]
```diff
@@ -4,18 +4,21 @@
...
     """
     Class will hold predefined naming of all classification ML Metrics supported by ParallelM.
+    When add new enum, please register stat creation function to ClassificationStatObjectFactory.registry_name_to_function.
     """
     ACCURACY_SCORE = "Accuracy Score"
     AUC = "AUC(Area Under the Curve)"
     AVERAGE_PRECISION_SCORE = "Average Precision Score"
     BALANCED_ACCURACY_SCORE = "Balanced Accuracy Score"
     BRIER_SCORE_LOSS = "Brier Score Loss"
+    CLASSIFICATION_REPORT = "Classification Report"
```
Comment (yakov-g) [inline mlops/parallelm/mlops/metrics_constants.py]: maybe you can iterate over enums and registered handlers and check if their number match

File: mlops/parallelm/mlops/ml_metrics_stat/classification/classification_stat_object_factory.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        It is not recommended to access this method without understanding single value data structure that it is returning.
+        :param kwargs: classification report string.
+        :return: Table Value stat object which has classification report embedded inside
+        """
+        cr = kwargs.get('data', None)
+
+        category = StatCategory.GENERAL
+
+        if cr is not None:
+
```
Comment (yakov-g) [inline mlops/parallelm/mlops/ml_metrics_stat/classification/classification_stat_object_factory.py]: remove empty line

File: mlops/parallelm/mlops/ml_metrics_stat/classification/classification_stat_object_factory.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    # registry holds name to function mapping. please add __func__ for making static object callable from below getter method.
+    registry_name_to_function = {
+        ClassificationMetrics.ACCURACY_SCORE: get_mlops_accuracy_score_stat_object.__func__,
+        ClassificationMetrics.AUC: get_mlops_auc_stat_object.__func__,
+        ClassificationMetrics.AVERAGE_PRECISION_SCORE: get_mlops_average_precision_score_stat_object.__func__,
+        ClassificationMetrics.BALANCED_ACCURACY_SCORE: get_mlops_balanced_accuracy_score_stat_object.__func__,
+        ClassificationMetrics.BRIER_SCORE_LOSS: get_mlops_brier_score_loss_stat_object.__func__,
+        ClassificationMetrics.CLASSIFICATION_REPORT: get_mlops_classification_report_stat_object.__func__,
+        ClassificationMetrics.CONFUSION_MATRIX: get_mlops_confusion_matrix_stat_object.__func__,
+    }
```
Comment (yakov-g) [inline mlops/parallelm/mlops/ml_metrics_stat/classification/classification_stat_object_factory.py:186]: iterate over enums and registry, or just check lengths

File: mlops/parallelm/mlops/ml_metrics_stat/ml_stat_object_creator.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            header = list(map(lambda x: str(x).strip(), list_2d[0]))
+            len_of_header = len(header)
+            table_object = Table().name(name).cols(header)
+
+            for index in range(1, len(list_2d)):
+                assert len(list_2d[index]) - 1 == len_of_header, \
+                    "length of row value does not match with headers lenght"
+
+                row_title = str(list_2d[index][0]).strip()
+
```
Comment (yakov-g) [inline mlops/parallelm/mlops/ml_metrics_stat/ml_stat_object_creator.py]: why empty lines?

**Review by yakov-g — APPROVED** [review]

File: mlops/parallelm/mlops/metrics_constants.py [hunk 5]
```diff
@@ -4,18 +4,21 @@
 class ClassificationMetrics(Enum):
     """
     Class will hold predefined naming of all classification ML Metrics supported by ParallelM.
+    When add new enum, please register stat creation function to ClassificationStatObjectFactory.registry_name_to_function.
```
Comment (DrewParallel) [inline mlops/parallelm/mlops/metrics_constants.py]: When adding a new enum, register a stat creation function to ...

File: mlops/parallelm/mlops/metrics_constants.py [hunk 6]
```diff
@@ -4,18 +4,21 @@
...
     BALANCED_ACCURACY_SCORE = "Balanced Accuracy Score"
     BRIER_SCORE_LOSS = "Brier Score Loss"
+    CLASSIFICATION_REPORT = "Classification Report"
     CONFUSION_MATRIX = "Confusion Matrix"
 
 
 class RegressionMetrics(Enum):
     """
     Class will hold predefined naming of all regression ML Metrics supported by ParallelM.
+    When add new enum, please register stat creation function to RegressionStatObjectFactory.registry_name_to_function.
```
Comment (DrewParallel) [inline mlops/parallelm/mlops/metrics_constants.py]: Ditto.

**Review by DrewParallel — CHANGES_REQUESTED** [review]

File: mlops/parallelm/mlops/metrics_constants.py (hunk 1 above)
Comment (harshilpm) [reply mlops/parallelm/mlops/metrics_constants.py]: i added that thing to my last pr: https://github.com/mlpiper/mlpiper/pull/95

**Comment by harshilpm** [comment]
closing pr
please review: https://github.com/mlpiper/mlpiper/pull/95

**Pull request closed without merging**